tests/fixtures/data.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2022 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+from pathlib import Path
+
+from omegaconf import DictConfig
+from pytest import fixture
+from torch.utils.data import DataLoader
+from transformers import PreTrainedTokenizerBase
+
+from llmfoundry.data.finetuning.dataloader import build_finetuning_dataloader
+from tests.data_utils import make_tiny_ft_dataset
+
+
+@fixture
+def tiny_ft_dataset_path(tmp_path: Path, dataset_size: int = 4) -> str:
+    """Creates a tiny dataset and returns the path."""
+    tiny_dataset_path = os.path.join(tmp_path, 'test-ift-data-small')
+    os.mkdir(tiny_dataset_path)
+    tiny_dataset_file = os.path.join(tiny_dataset_path, 'train.jsonl')
```

Comment:
Sort of a nit, but paths shouldn't require os at all
```suggestion
    tiny_dataset_path = tmp_path / 'test-ift-data-small'
    tiny_dataset_path.mkdir()
    tiny_dataset_file = tiny_dataset_path / 'train.jsonl'
```